src/main/scala/beam/router/RoutingWorker.scala

```diff
@@ -385,9 +385,15 @@ class RoutingWorker(workerParams: R5Parameters) extends Actor with ActorLogging
       val definedResponses = responses.flatten
       (definedResponses, r5ResponseOption) match {
         case (head +: _, Some(r5Resp)) =>
-          head.copy(itineraries = definedResponses.flatMap(_.itineraries) ++ r5Resp.itineraries)
+          head.copy(
+            itineraries = definedResponses.flatMap(_.itineraries) ++ r5Resp.itineraries,
+            searchedModes = definedResponses.map(_.searchedModes).reduce(_ ++ _) ++ r5Resp.searchedModes
```

Comment:
Implemented your suggestion, but I used flatMap instead of map..flatten as my IDE suggested. And because reduce(_ ++ _) produces a set I also swapped operators of ++